### PR TITLE
Implement JITM dismiss api 

### PR DESCRIPTION
**Pull request by AnirudhBhat**
### Note: Please make sure [this WooCommerce PR](https://github.com/woocommerce/woocommerce-android/pull/7653) is reviewed and ready to merge before merging this PR.

This PR implements JITM dismiss API. 

`POST` request body: `{"id":"woomobile_ipp_barcode_users"}`
`POST` request path: `jetpack/v4/jitm`

### CODE REVIEW

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmDismissApiResponse.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.wordpress.android.fluxc.network.rest.wpcom.wc.jitm
+
+import com.google.gson.annotations.SerializedName
+
+data class JitmDismissApiResponse(
```
Comment (kidinov) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmDismissApiResponse.kt]: Not used?

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt [hunk 2]
```diff
@@ -50,4 +50,36 @@ class JitmRestClient @Inject constructor(
...
+
+        val response = jetpackTunnelGsonRequestBuilder.syncPostRequest(
+            this,
+            site,
+            url,
+            mapOf(
+                "id" to jitmId,
+                "feature_class" to featureClass
+            ),
+            Any::class.java
```
Comment (kidinov) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt:69]: Shell it be `JitmDismissApiResponse` here?

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt [hunk 3]
```diff
@@ -50,4 +50,36 @@ class JitmRestClient @Inject constructor(
...
+        )
+
+        return when (response) {
+            is JetpackTunnelGsonRequestBuilder.JetpackResponse.JetpackSuccess -> {
+                WooPayload(true)
```
Comment (kidinov) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt:74]: If the plan was to use `JitmDismissApiResponse` then changes here needed as well

**Review by kidinov — COMMENTED** [review]
LGTM. Just want to make sure if `JitmDismissApiResponse` is something you forgot to use or forgot to remove?

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt (hunk 3 above)
Comment (kidinov) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt:74]: Also just checking, can it response like that?

```
{
  "data": false
}
```

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmDismissApiResponse.kt (hunk 1 above)
Comment (AnirudhBhat) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmDismissApiResponse.kt]: Yes. Forgot to remove it! Deleted [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2552/commits/089e51df7d12b6f96996196c3d8c6ca4c7b70309). Thanks!

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt (hunk 3 above)
Comment (AnirudhBhat) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/jitm/JitmRestClient.kt:74]: AFAIK, it either returns `"data" : true` or throws error with status code other than 200

**Review by kidinov — APPROVED** [review]
👍

**Pull request merged**